sdds/writer.py

```diff
@@ -51,15 +51,23 @@ def _sdds_def_as_str(definition: Union[Description, Definition, Data]) -> str:
 def _write_data(names: List[str], sdds_file: SddsFile, outbytes: IO[bytes]) -> None:
     # row_count:
     outbytes.write(np.array(0, dtype=get_dtype_str("long")).tobytes())
-    _write_parameters((sdds_file[name] for name in names
-                       if isinstance(sdds_file.definitions[name], Parameter)),
-                      outbytes)
-    _write_arrays((sdds_file[name] for name in names
-                   if isinstance(sdds_file.definitions[name], Array)),
-                  outbytes)
-    _write_columns((sdds_file[name] for name in names
-                    if isinstance(sdds_file.definitions[name], Column)),
-                   outbytes)
+
+    def __getitem__(self, name: str) -> Tuple[Definition, Any]:
+        return self.definitions[name], self.values[name]
+
+    parameters: List[Any] = []
+    arrays: List[Any] = []
+    columns: List[Any] = []
```

Comment:
I think these could be typed a little more specifically since we know what would be appended (we specifically check against `Parameter`, `Array` or `Column` types) depending on the if clause triggered.

Reply:
I didn't know those things about poetry. I have only seen it working on projects with very large number of dependencies. I liked poetry because it is easy to manage virtual environments. With pip, it was always a mess.